test/subscription.ts

```diff
@@ -596,123 +576,103 @@ describe('Subscription', () => {
   });
 
   describe('get', () => {
-    beforeEach(() => {
-      subscription.create = util.noop;
-    });
-
     it('should delete the autoCreate option', done => {
       const options = {
         autoCreate: true,
         a: 'a',
       };
-
-      subscription.getMetadata = (gaxOpts: CallOptions) => {
+      sandbox.stub(subscription, 'getMetadata').callsFake((gaxOpts) => {
         assert.strictEqual(gaxOpts, options);
         // tslint:disable-next-line no-any
-        assert.strictEqual((gaxOpts as any).autoCreate, undefined);
+        assert.strictEqual((gaxOpts as typeof options).autoCreate, undefined);
         done();
-      };
+      });
+
 
       subscription.get(options, assert.ifError);
     });
 
     describe('success', () => {
       const fakeMetadata = {};
 
-      beforeEach(() => {
-        subscription.getMetadata =
-            (gaxOpts: CallOptions,
-             callback: GetSubscriptionMetadataCallback) => {
-              callback(null, fakeMetadata);
-            };
-      });
-
       it('should call through to getMetadata', done => {
-        subscription.get(
-            (err: Error, sub: Subscription,
-             resp: google.pubsub.v1.ISubscription) => {
-              assert.ifError(err);
-              assert.strictEqual(sub, subscription);
-              assert.strictEqual(resp, fakeMetadata);
-              done();
+        sandbox.stub(subscription, 'getMetadata')
+            .callsFake((gaxOpts, callback) => {
+              callback(null, fakeMetadata);
             });
+
+        subscription.get((err, sub, resp) => {
+          assert.ifError(err);
+          assert.strictEqual(sub, subscription);
+          assert.strictEqual(resp, fakeMetadata);
+          done();
+        });
       });
 
       it('should optionally accept options', done => {
         const options = {};
-
-        subscription.getMetadata =
-            (gaxOpts: CallOptions,
-             callback: GetSubscriptionMetadataCallback) => {
+        sandbox.stub(subscription, 'getMetadata')
+            .callsFake((gaxOpts, callback) => {
               assert.strictEqual(gaxOpts, options);
               callback(null);  // the done fn
-            };
+            });
 
         subscription.get(options, done);
       });
     });
 
     describe('error', () => {
+      afterEach(() => sandbox.restore());
```

Comment:
I'd prefer we pull this up way to the top, and do this in `afterEach` for ever test in the file.

Reply:
I think this already exists in a global hook

Reply:
Done